org-code-javabuilder/media/src/main/java/org/code/media/Image.java

```diff
@@ -156,17 +157,39 @@ public BufferedImage getBufferedImage() {
    * @throws FileNotFoundException if the file is not found
    */
   public static BufferedImage getImageAssetFromFile(String filename) throws FileNotFoundException {
+    BufferedImage originalImage;
     try {
-      BufferedImage image =
+      originalImage =
           ImageIO.read(new URL(GlobalProtocol.getInstance().generateAssetUrl(filename)));
-      if (image == null) {
+      if (originalImage == null) {
         // this can happen if the filename is not associated with an image
         throw new MediaRuntimeException(MediaRuntimeExceptionKeys.IMAGE_LOAD_ERROR);
       }
-      return image;
     } catch (IOException e) {
       throw new FileNotFoundException("Could not find file " + filename);
     }
+
+    // Resize the image to a max size of 400x400px while maintaining the aspect ratio.
+    // This allows us to save time when students run on pixel manipulation on an entire image.
+    int height = originalImage.getHeight();
+    int width = originalImage.getWidth();
+    if (width <= 400 && height <= 400) {
+      return originalImage;
+    }
+    int targetHeight, targetWidth;
+    if (height > width) {
+      targetHeight = 400;
+      targetWidth = (int) ((double) width / height * 400);
+    } else {
+      targetWidth = 400;
+      targetHeight = (int) ((double) height / width * 400);
+    }
+    BufferedImage image = new BufferedImage(targetWidth, targetHeight, BufferedImage.TYPE_INT_RGB);
```

Comment:
I think we might want this to be of type `TYPE_INT_ARGB` so we can maintain opacity

Reply:
👍